crates/sui-indexer-alt-graphql/src/pagination.rs

```diff
@@ -290,9 +290,77 @@ pub(crate) fn is_connection(field: &MetaField) -> bool {
     type_.ends_with("Connection") || type_.ends_with("Connection!")
 }
 
+/// Ensures ASC order DESC order, then applies pagination limits with overhead for paginate_results to check
+/// for has_prev and has_next page. Returns items in ascending order as GraphQL cursor syntax expects items
+/// to be in ascending order.
+///
+/// ```
+/// // Example usage:
+/// let events: Vec<Event> = /* ... */;
+/// let results = ordered_items_with_limits(
+///     &page,
+///     events,
+///     |e| (e.tx_sequence_number, e.sequence_number)
+/// );
+/// ```
+pub(crate) fn ordered_results_with_limits<Result, SortKey, KeyFn, Cursor>(
+    page: &Page<Cursor>,
+    results: impl IntoIterator<Item = Result>,
+    key_fn: KeyFn,
+) -> Vec<Result>
+where
+    SortKey: Ord,
+    KeyFn: Fn(&Result) -> SortKey,
+{
+    let mut results: Vec<Result> = results.into_iter().collect();
+
+    if results.is_empty() {
+        return results;
+    }
+
+    // Ensures items are in ASC or DESC order based on page.is_from_front().
+    results.sort_by(|a, b| {
+        let key_a = key_fn(a);
+        let key_b = key_fn(b);
+
+        if page.is_from_front() {
+            key_a.cmp(&key_b)
+        } else {
+            key_b.cmp(&key_a)
+        }
+    });
```

Comment:
It would be better to pull the `is_from_front` check outside the call to `sort_by` to avoid re-testing this condition before every comparison, but even more than that, it would better to avoid using `sort_by` altogether -- the results into this function should already be essentially sorted, so we should be able to just call either just `truncate` or `reverse; truncate; reverse` on it, unless I'm missing something (which probably does not warrant its own function).